src/components/UserTap.js

```diff
@@ -0,0 +1,133 @@
+import { useRef, useState } from "react";
+import styled from "styled-components";
+
+import { postData } from "../api/service";
+
+import MESSAGES from "../constants/messages";
+import validateNickname from "../utils/validateNickname";
+import Button from "./common/Button";
+import addPhoto from "../addPhoto";
+import ProfileImage from "./RegisterCard/ProfileImage";
+
+const {
+  FAILED_UPLOAD_IMAGE,
+  USER_INFORMATION_UPDATED,
+  OK,
+} = MESSAGES;
+
+const Wrapper = styled.div`
+  position: absolute;
+  z-index: 1;
+  width: 430px;
+  height: 100%;
+  box-shadow: 10px 10px 10px rgba(0, 0, 0, 0.25);
+`;
+
+const BlankBlock = styled.div`
+  width: 430px;
+  height: 100px;
+`;
+
+const Information = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  width: 430px;
+  height: 200px;
+  margin: 80px 0;
+  text-align: center;
+`;
+
+const NickName = styled.div`
+  font-weight: bold;
+  font-size: 35px;
+`;
+
+const Input = styled.input`
+  display: block;
+  width: 230px;
+  height: 30px;
+  padding-left: 10px;
+  margin-left: 90px;
+  border: 2px solid #543FD3;
+  border-radius: 4px;
+  outline: none;
+  transition: 0.2s;
+
+  &:focus {
+    border: 2px solid #26BFA6;
+  }
+
+  &::placeholder {
+    font-size: 14px;
+  }
+`;
+
+const Email = styled.div`
+  font-size: 25px;
+`;
+
+const SubscriberNumber = styled.div`
+  font-size: 25px;
+`;
+
+const ButtonWrapper = styled.div`
+  margin-left: 160px;
+  margin-bottom: 100px;
+`;
+
+const Message = styled.p`
+  font-size: 12px;
+  color: var(--color-message);
+`;
+
+export default function UserTap({ user }) {
+  const { email, nickname, imageUrl, subscriberList } = user;
+
+  const [failureReason, setFailureReason] = useState();
+  const reference = useRef();
+
+  const handleButtonClick = async () => {
+    const nickname = reference.current.value;
+
+    const validationResult = validateNickname(nickname);
+
+    if (validationResult !== OK) {
+      setFailureReason(validationResult);
+
+      return;
+    }
+
+    addPhoto()
+      .then(async (data, error) => {
+        if (error) {
+          return alert(FAILED_UPLOAD_IMAGE);
```

Comment:
`window.alert()`는 실행 결과로 `undefined`를 반환합니다. 따라서 다음과 같이 작성하면 좋을 것 같습니다!

```suggestion
          alert(FAILED_UPLOAD_IMAGE);
          return;
```

Reply:
말씀해주신 사항 모두 반영하였습니다!!